bin/webpack-dev-server.js

```diff
@@ -418,7 +407,42 @@ function startDevServer(config, options) {
 
       status(uri, options, log, argv.color);
     });
+  } else {
+    // only run port finder if no port as been specified
+    portRetries = 0;
+    portfinder.basePort = DEFAULT_PORT;
+    server.listeningApp.on('error', (err) => {
+      if (err.code === 'EADDRINUSE' && portRetries < DEFAULT_PORT_RETRY) {
+        tryStartListening(options, suffix, log);
+        return;
+      }
+      throw err;
+    });
+    tryStartListening(options, suffix, log);
   }
 }
 
+function tryStartListening(options, suffix, log) {
+  portRetries += 1;
+  portfinder.getPort((err, port) => {
```

Comment:
Need create `utils` what get port, no need move `listen` logic in this utils

Reply:
@u9520107 we need split the logic of searching free port and starting webserver, also you code doesn't solve problem second port also can be busy

Reply:
I'll see what i can do to separate them. But to your second comment, if the second port is busy as well, it will throw an error with code==='EADDRINUSE' again, then trigger another get port. This modification will allow an instance to try at most 10 times before giving up.

Reply:
@u9520107 let's use `process.env.DEFAULT_PORT_RETRY` to allow people setup this, also let's do it by default `3`

Reply:
OK!